### ISSUE TITLE
Transmitting fails with Device(Unknown) on raspberry pi

### ISSUE THREAD

**Issue by ShadowJonathan** [28757B2/cc1101-rust#1]
This is the same context and probably problem as https://github.com/28757B2/cc1101-driver/issues/4

Using the following code, I get a Device(Unknown) problem when trying to transmit something

<details>
<summary>Code</summary>

```rs
use cc1101_rust::{
    config::{Modulation, TXConfig},
    CC1101Error, CC1101,
};
const ONE: [u8; 16] = [
    0x55, 0x55, 0x55, 0x53, 0x33, 0x54, 0xd5, 0x35, 0x55, 0x54, 0xac, 0xad, 0x00, 0x00, 0x00, 0x00,
];

const THREE: [u8; 16] = [
    0x55, 0x55, 0x55, 0x53, 0x33, 0x54, 0xd5, 0x35, 0x55, 0x4c, 0xca, 0xb5, 0x00, 0x00, 0x00, 0x00,
];

fn main() -> Result<(), CC1101Error> {
    let tx_config = TXConfig::new(
        868.384,
        Modulation::FSK2,
        8.192,
        0.6,
        Some(20.629883),
        Some(0xAAAB),
    )?;
    let cc1101 = CC1101::new("/dev/cc1101.0.0", None, false)?;

    cc1101.transmit(&tx_config, &THREE)?;

    Ok(())
}
```

</details>

This is when using 1.3.3 of the cc1101 driver

**Comment by 28757B2** [comment]
Thanks for the issue report. This is another mismatch between the struct size in the kernel and userspace.

I think I'm going to address this in the driver by fixing the type widths. That should remove all of these 32/64-bit struct size issues. The Rust implementation will then become correct and the Python version can be updated to match.

**Comment by ShadowJonathan** [comment]
Thank you so much for working on this :) ❤️

**Comment by 28757B2** [comment]
This is fixed in driver 1.4.0. You'll need version 0.2.1 of the Rust crate to talk to the driver, as I've bumped the protocol version because of the changes to the IOCTL structs.

**Comment by ShadowJonathan** [comment]
Thank you very much, I'll test these changes as soon as I have access to another RPI again, as the one I've been using up until now is now assigned for another task.
